camp/jbolotski/beam_break_motor/src/main/java/frc/robot/subsystems/BeamBreakSensor.java

```diff
@@ -0,0 +1,65 @@
+package frc.robot.subsystems;
+
+import edu.wpi.first.math.filter.Debouncer.DebounceType;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj2.command.Subsystem;
+import edu.wpi.first.wpilibj2.command.button.Trigger;
+import frc.robot.Constants;
+import frc.robot.common.Types.BeamBreakState;
+
+/**
+ * Creates a Trigger from a DigitalInput for a Beam Break sensor
+ * returns true when the beam is unbroken
+ */
+public class BeamBreakSensor extends Trigger{
```

Comment:
I think https://first.wpi.edu/wpilib/allwpilib/docs/release/java/edu/wpi/first/wpilibj2/command/button/Trigger.html might help visualize a bit more about the subclass here.  The way I see it, we really only need to extend a single method (plus have a default ctor) for the situation at hand in this project.

Reply:
There were two things that were keeping me from directly building the Trigger from the DigitalInput in this straight-forward way:
1) when you do that it's not clear anywhere in the code that DigitalInput =true represents a broken or unbroken beam.  I would like for someone reading the top level code to be able to tell that the motor runs when the beam is unbroken and stops when the beam is broken.   
2) it forces you to call the debouncer at the top-level.  The debouncer method does not cause the current trigger to be debounced, it returns a whole new trigger.  I didn't want the call to the debouncer at the top level, because, I feel like it's an implementation detail that the top-level user shouldn't need to know.